bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -875,6 +881,20 @@ public void run() {
     @Override
     public void modifiedService(ServiceReference/* <ModuleHandlerFactory> */ reference,
             /* ModuleHandlerFactory */Object service) {
+        logger.debug("ModuleHandlerFactory modified, updating handlers");
+        ModuleHandlerFactory moduleHandlerFactory = ((ModuleHandlerFactory) service);
+        @SuppressWarnings("unchecked")
+        Collection<String> removed_handlers = (Collection<String>) reference.getProperty("removed_handlers");
```

Comment:
May be it can be done: 
1) iterate through the  moduleHandlerFactories to get all old module type registered by this factory (oldTypeList)
2) oldTypeList - newTypes = removedTypes
3) newTypes - oldTypeList = addedTypes
But in any case we must have registration property which has to be changed (otherwise the modified service will not be called). 
I don't see a real use case where the types of factory have to be changed runtime (without restart of the factory).